assembl/static2/js/app/components/debate/thread/announcement.jsx

```diff
@@ -0,0 +1,91 @@
+import React from 'react';
+import { Translate } from 'react-redux-i18n';
+import { Col, Tooltip } from 'react-bootstrap';
+
+import StatisticsDoughnut from '../common/statisticsDoughnut';
+import { sentimentDefinitionsObject } from './sentimentDefinitions';
+import Video from './video';
+import { PublicationStates } from '../../../constants';
+
+const createTooltip = (sentiment, count) => {
+  return (
+    <Tooltip id={`${sentiment.camelType}Tooltip`} className="no-arrow-tooltip">
+      {count} <Translate value={`debate.${sentiment.camelType}`} />
+    </Tooltip>
+  );
+};
+
+const getSentimentsCount = (posts) => {
+  const counters = { ...sentimentDefinitionsObject };
+  Object.keys(counters).forEach((key) => {
+    counters[key].count = 0;
+  });
+  posts.edges.forEach(({ node: { sentimentCounts, publicationState } }) => {
+    if (publicationState === PublicationStates.PUBLISHED) {
+      Object.keys(counters).forEach((key) => {
+        counters[key].count += sentimentCounts[key];
+      });
+    }
+  });
+  return counters;
+};
+
+const createDoughnutElements = (sentimentCounts) => {
+  return Object.keys(sentimentCounts).map((key) => {
+    return {
+      color: sentimentCounts[key].color,
+      count: sentimentCounts[key].count,
+      Tooltip: createTooltip(sentimentCounts[key], sentimentCounts[key].count)
+    };
+  });
+};
+
+const dirtySplitHack = (announcementBody) => {
+  // To allow edit from V1 announcement, add !split!https://video.url!split!
+  const split = announcementBody.split('!split!');
```

Comment:
Yeah, so we described this earlier today. Now I see that you're using the actual split function. What I would recommend to make your hack more functional is to use regex groups.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp#grouping-back-references

Reply:
what are the advantages of the regexp?